### PR TITLE
Hw 02/layout

### PR DESCRIPTION
**Pull request by tunya**

### CODE REVIEW

File: app/src/components/FilterItem/FilterItem.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
+@import "../../shared/variables";
+
+.filter_item {
```
Comment (Aksana-Tsishchanka) [inline app/src/components/FilterItem/FilterItem.scss]: If you use CSS Modules, there are not necessary to use BEM, check https://css-tricks.com/css-modules-part-1-need/#article-header-id-4

File: app/src/components/Footer/Footer.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import PropTypes from 'prop-types';
+import styles from './Footer.scss';
+
+function Footer(props) {
+  return (
+    <footer className={styles.footer}>{props.text}</footer>
+  );
+}
+
+Footer.propTypes = {
```
Comment (Aksana-Tsishchanka) [inline app/src/components/Footer/Footer.jsx:11]: nice

File: app/src/components/Header/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Search from './../Search/Search';
+import Movie from './../Movie/Movie';
+import styles from './Header.scss';
+
+function Header(props) {
+  let content = '';
```
Comment (Aksana-Tsishchanka) [inline app/src/components/Header/Header.jsx]: I think ternary operator is more suitable here http://take.ms/V0lyN

File: app/src/components/Header/Header.scss [hunk 4]
```diff
@@ -0,0 +1,37 @@
+@import "../../shared/variables";
+.header {
+  background: url('./img/bg.jpg');
+
+  h1 {
```
Comment (Aksana-Tsishchanka) [inline app/src/components/Header/Header.scss]: Avoid using a selector with an element, it does not make sense for CSS Modules. If you want to get all advantages of CSS Modules use classes and do style more independent. H1 depends on a parent element, it's not right for CSS Modules. Keep styles simple.

File: app/src/components/Movie/Movie.jsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+          <p className={styles.hero_crew}>
+            Cast: {item.show_cast}
+          </p>
+        </section>
+      </main>
+    );
+  }
+
+  renderListed() {
+    const item = this.props.item;
```
Comment (Aksana-Tsishchanka) [inline app/src/components/Movie/Movie.jsx]: destructuring to get show_title and so on?

File: app/src/components/Movie/Movie.scss [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  cursor: pointer;
+  border: 1px solid #fff;
+  border-radius: 5px;
+  transition: border-color .3s ease;
+
+  &:hover {
+    border-color: #C7C7C7;
+  }
+
+  img {
```
Comment (Aksana-Tsishchanka) [inline app/src/components/Movie/Movie.scss]: avoid elements in selectors

File: app/src/components/App/App.jsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+export default class App extends React.Component {
+  static sortListBy(list, criteria) {
+    return _.reverse(_.sortBy(list, criteria));
+  }
+
+  constructor() {
+    super();
+    this.state = {
+      movies: response,
```
Comment (Aksana-Tsishchanka) [inline app/src/components/App/App.jsx]: I recommend operating with response in a separate component, for ex: MovieListContainer.
App component has to pass props which affect on movie list and when props changes, the movie list has to change as well. Check again https://vasanthk.gitbooks.io/react-bits/patterns/25.presentational-vs-container.html (consider thoroughly the code in componentDidMount(), it's the place where you have to request data)

File: app/src/components/App/App.jsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+      movies = App.sortListBy(movies, this.state.sortOrder);
+    }
+    this.setState({
+      query: e.target.query.value,
+      movies,
+    });
+    return false;
+  }
+
+  handleSortBy({ target: { value } }) {
```
Comment (Aksana-Tsishchanka) [inline app/src/components/App/App.jsx:41]: nice, you can use it everywhere)

File: app/src/components/FilterItem/FilterItem.scss (hunk 1 above)
Comment (tunya) [reply app/src/components/FilterItem/FilterItem.scss]: Done

File: app/src/components/Header/Header.jsx (hunk 3 above)
Comment (tunya) [reply app/src/components/Header/Header.jsx]: Done

File: app/src/components/Header/Header.scss (hunk 4 above)
Comment (tunya) [reply app/src/components/Header/Header.scss]: Fixed

File: app/src/components/Movie/Movie.jsx (hunk 5 above)
Comment (tunya) [reply app/src/components/Movie/Movie.jsx]: Done

File: app/src/components/Movie/Movie.scss (hunk 6 above)
Comment (tunya) [reply app/src/components/Movie/Movie.scss]: Fixed

File: app/src/components/App/App.jsx (hunk 7 above)
Comment (tunya) [reply app/src/components/App/App.jsx]: Implemented.

File: app/src/components/MovieListContainer/MovieListContainer.jsx [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  constructor() {
+    super();
+    this.state = {
+      movies: [],
+    };
+  }
+
+  componentDidMount() {
+    // TODO: write an API request, wrap everything in a promise
+    this.getMovies();
```
Comment (Aksana-Tsishchanka) [inline app/src/components/MovieListContainer/MovieListContainer.jsx:21]: I did not get it. When Promise will be resolved, you have to setState with received movies. SO now you can do: `this.setState({movies: response}`;

Comment (tunya) [reply app/src/components/MovieListContainer/MovieListContainer.jsx:21]: That's done for 2 reasons:

1. I'll need to get another list of movies once the search parameters are updated, so separate reusable method made sense to me. 
2. Eslint throws an error when it sees this.setState call in the .componentDidMount method.

**Pull request merged**